src/main/kotlin/com/chattriggers/ctjs/utils/console/ElementaConsole.kt

```diff
@@ -137,6 +143,28 @@ class ConsoleComponent(private val loader: ILoader?) : WindowScreen(ElementaVers
                 textInput.setText("")
                 scrollToLastInput()
             }
+
+            textInput.onKeyType { _, keyCode ->
+                if (keyCode == GLFW.GLFW_KEY_UP && Client.isControlDown()) {
+                    historyOffset++
+                    try {
+                        val message = history[history.size - historyOffset]
+                        textInput.setText(message)
+                    } catch (exception: Exception) {
+                        historyOffset--
+                    }
+                } else if (keyCode == GLFW.GLFW_KEY_DOWN && Client.isControlDown()) {
+                    historyOffset = maxOf(0, historyOffset - 1)
```

Comment:
I usually prefer Kotlin's `coerce` functions for things like this
```suggestion
                    historyOffset = (historyOffset - 1).coerceAtLeast(0)
```